### PR TITLE
feat(versions): Add support for deep linking to ContentSidebar

### PR DESCRIPTION
**Pull request by jstoffan**
Todo
- [x] Unit tests

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/Sidebar.js [hunk 1]
```diff
@@ -162,6 +169,20 @@ class Sidebar extends React.Component<Props, State> {
...
+     * @param {string} path - The path to navigate
+     * @param {Object} options - Optional options object
+     * @param {boolean} options.replace - Optional flag to replace the current path
+     * @returns void
+     */
+    navigate = (path: string = '/', { replace = false }: { replace: boolean } = {}) => {
+        const { history } = this.props;
+        const method = replace ? history.replace : history.push;
+
+        method(path.charAt(0) === '/' ? path : `/${path}`);
```
Comment (jeremypress) [inline src/elements/content-sidebar/Sidebar.js]: Are other places calling navigate that don't include a `/`? Both calls above seem to.

File: src/elements/content-sidebar/Sidebar.js [hunk 2]
```diff
@@ -162,6 +169,20 @@ class Sidebar extends React.Component<Props, State> {
...
     }
 
+    /**
+     * Helper to navigate sidebar tabs programatically
+     * @param {string} path - The path to navigate
+     * @param {Object} options - Optional options object
+     * @param {boolean} options.replace - Optional flag to replace the current path
+     * @returns void
+     */
+    navigate = (path: string = '/', { replace = false }: { replace: boolean } = {}) => {
```
Comment (jeremypress) [inline src/elements/content-sidebar/Sidebar.js]: Do we want to switch all `history.pushes` to use navigate?

File: src/elements/content-sidebar/Sidebar.js [hunk 3]
```diff
@@ -81,11 +83,17 @@ class Sidebar extends React.Component<Props, State> {
...
 
     componentDidUpdate(prevProps: Props): void {
-        const { fileId, history, isLarge, location }: Props = this.props;
-        const { fileId: prevFileId, isLarge: prevIsLarge }: Props = prevProps;
+        const { defaultPath, fileId, history, isLarge, location }: Props = this.props;
+        const { defaultPath: prevDefaultPath, fileId: prevFileId, isLarge: prevIsLarge }: Props = prevProps;
         const { isDirty, isOpen }: State = this.state;
         const isForcedSet = this.isForcedSet();
 
+        // Force navigate if host application insists
```
Comment (jeremypress) [inline src/elements/content-sidebar/Sidebar.js]: add this case to the componentDidUpdate UT?

Comment (jstoffan) [reply src/elements/content-sidebar/Sidebar.js]: Done.

File: src/elements/content-sidebar/Sidebar.js [hunk 4]
```diff
@@ -82,13 +82,19 @@ class Sidebar extends React.Component<Props, State> {
 
     componentDidUpdate(prevProps: Props): void {
         const { fileId, history, isLarge, location }: Props = this.props;
-        const { fileId: prevFileId, isLarge: prevIsLarge }: Props = prevProps;
+        const { fileId: prevFileId, isLarge: prevIsLarge, location: prevLocation }: Props = prevProps;
         const { isDirty, isOpen }: State = this.state;
+        const { state = {} } = location;
```
Comment (jeremypress) [inline src/elements/content-sidebar/Sidebar.js]: Maybe we should rename this as to not confuse it with the component state?

File: src/elements/content-sidebar/SidebarPanels.js [hunk 5]
```diff
@@ -85,87 +83,97 @@ const SidebarPanels = ({
...
-                        parentName={match.params.sidebar}
-                        versionId={match.params.versionId}
-                    />
-                )}
+                    )}
+                />
+            )}
+            {hasVersions && (
+                <Route
+                    path="/:sidebar/versions/:versionId?"
```
Comment (jeremypress) [inline src/elements/content-sidebar/SidebarPanels.js]: You mentioned moving to a model where a version is always selected. Do we we need to remove this optionality?

Comment (jstoffan) [reply src/elements/content-sidebar/SidebarPanels.js]: Not yet, unfortunately. We don't always have access to the current version id in the host application at the moment, so there will be some cases where it's empty. I'd like to improve that in the near future, though.

File: src/elements/content-sidebar/Sidebar.js (hunk 4 above)
Comment (jstoffan) [reply src/elements/content-sidebar/Sidebar.js]: Sure, no problem.

File: src/elements/content-sidebar/SidebarRouter.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import * as React from 'react';
+import { MemoryRouter, Router } from 'react-router-dom';
+import type { RouterHistory } from 'react-router-dom';
+
+type Props = {
+    children: React.Node,
+    history?: RouterHistory,
+};
+
+const SidebarRouter = ({ children, history, ...rest }: Props) => {
```
Comment (j3tan) [inline src/elements/content-sidebar/SidebarRouter.js]: Will this eventually be at the ContentPreview level instead of just the Sidebar?

**Comment by j3tan** [comment]
code looks good, pretty straightforward to understand

File: src/elements/content-sidebar/SidebarRouter.js (hunk 6 above)
Comment (jstoffan) [reply src/elements/content-sidebar/SidebarRouter.js]: That's the ultimate goal, but it'll require top-level routes in our host application.

**Pull request merged**